inference-engine/src/mkldnn_plugin/nodes/mkldnn_concat_node.cpp

```diff
@@ -27,291 +27,307 @@ using namespace mkldnn;
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
 
+
+bool MKLDNNConcatNode::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        errorMessage = std::string("CPU plugin doesn't support nGraph operation ") + op->get_type_name() + " with name '" + op->get_friendly_name() + "' ";
+        auto concatOp = ngraph::as_type_ptr<const ngraph::op::v0::Concat>(op);
+        if (!concatOp) {
+            return false;
+        }
+        if (concatOp->get_input_size() == 0) {
+            errorMessage += "and inputs num " + std::to_string(concatOp->get_input_size()) + ". Two inputs expected.";
+            return false;
+        }
+        if (concatOp->get_output_size() != 1) {
```

Comment:
This condition doesn't check supported/unsupported cases. It checks does the operation corresponds specification or not. In this case it shouldn't be places inside isSupportedOperation();
The same for `if (concatOp->get_input_size() == 0) {
`

Reply:
In general number of I/O can be a criteria to execute transformation. I suggest to keep it in this function for all layers.

Reply:
As was discussed offline, we relies on nGraph in part of operation validation.